xrpl/asyncio/transaction/reliable_submission.py

```diff
@@ -113,3 +118,110 @@ async def send_reliable_submission(
     return await _wait_for_final_transaction_outcome(
         transaction_hash, client, prelim_result, 0
     )
+
+
+def _is_signed(transaction: Transaction) -> bool:
+    """
+    Checks if a transaction has been signed.
+
+    Args:
+        transaction: the transaction to check.
+
+    Returns:
+        Whether the transaction has been signed
+    """
+    return (
+        transaction.signing_pub_key is not None and len(transaction.signing_pub_key) > 0
+    ) or (transaction.txn_signature is not None and len(transaction.txn_signature) > 0)
+
+
+def _decode_tx_blob(tx_blob: str) -> Transaction:
+    """
+    Decodes a transaction blob.
+
+    Args:
+        tx_blob: the tx blob to decode.
+
+    Returns:
+        The formatted transaction.
+    """
+    return Transaction.from_xrpl(decode(tx_blob))
+
+
+async def _get_signed_tx(
+    transaction: Union[Transaction, str],
+    client: Client,
+    wallet: Optional[Wallet] = None,
+    check_fee: bool = True,
+    autofill: bool = True,
+) -> Transaction:
+    """
+    Initializes a signed transaction for a submit request.
+
+    Args:
+        transaction: the signed/unsigned transaction (or transaction blob) to
+            be submitted.
+        client: the network client with which to submit the transaction.
+        wallet: the wallet with which to sign the transaction (optional, only needed
+        if the transaction is not signed).
+        check_fee: an optional bolean indicating whether to check if the fee is
+            higher than the expected transaction type fee. Defaults to True.
+        autofill: an optional boolean indicating whether to autofill the
+            transaction. Defaults to True.
+
+    Returns:
+        The signed transaction.
+    """
+    if isinstance(transaction, str):
+        transaction = _decode_tx_blob(transaction)
+
+    if _is_signed(transaction):
+        return transaction
+
+    if not wallet:
+        raise XRPLException(
+            "Wallet must be provided when submitting an unsigned transaction"
+        )
```

Comment:
I'm not clear why `Wallet` is optional - it seems like instead of throwing an error, I think it'd be better to just require that a wallet be provided to use this function. 

There's no part of the code which takes advantage of the fact that you can just not pass in a `Wallet` and the case where you don't pass in a wallet, this function basically doesn't do anything - because either the transaction is already signed (no effect) or this function throws an exception.

Reply:
Requiring the wallet also makes this function essentially unusable by someone using a different method (e.g. XUMM) to sign their transactions.

Reply:
In practice, this function does require a wallet to do it's job. I'm also specifically talking about `_get_signed_tx` in this case, which is a private helper function, and so shouldn't be a problem for people using a different method since they won't call it. (Or maybe I'm misunderstanding the concern?)

Reply:
The current architecture of the `submit_and_wait` method involves everything going through `_get_signed_tx`, even already-signed transactions. So that's certainly doable, but it would involve some reworking.

Reply:
Both `submit_and_wait` and `_get_signed_tx` in xrpl.js have `Wallet` as an optional param, only required when the transaction is unsigned. I think updating the docstring to notify user and having this error message is adequate, but I'm open to suggestions.

Reply:
This seems fine to me for a private helper. I originallly thought you could just move the exception up out of the function and get the same behavior without a parameter, but now I realize you need to decode it first, and check the signature, which are decently helpful to have in this function :)